src/App.tsx

```diff
@@ -1,12 +1,81 @@
 import React from 'react';
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { GoodsList } from './components/GoodsList/GoodsList';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+import { getAll, get5First, getRedGoods } from './api/goods';
+
+type State = {
+  goods: Good[];
+  isLoading: boolean;
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [],
+    isLoading: false,
+  };
+
+  loadData = async (getData: () => Promise<Good[]>) => {
+    this.setState({
+      isLoading: true,
+    });
+
+    try {
+      const goods = await getData();
+
+      this.setState({
+        goods,
+        isLoading: false,
+      });
+    } catch {
+      this.setState({
+        isLoading: false,
+      });
+    }
```

Comment:
You should add some type of error message here and use conditional rendering for displaying it